src/topp/FileFilter.cpp

```diff
@@ -135,6 +135,69 @@ class TOPPFileFilter :
   }
 
 private:
+  static FeatureMap makeUnique(const FeatureMap& feature_map) 
+  {
+    // Flag: `f_and_c:make_unique`
+    
+    // This function does two things:
+    // 1. create a std::map `feature_set` that maps pairs of <charge, sequence> 
+    // to a pointer to the feature with the highest intensity for this sequence.
+    // 2. construct a FeatureMap that contains all unannotated features in `feature_map`
+    // and all features that are in `feature_set`.
+    
+    // Copy (unique) features from `feature_set` over into a fresh FeatureMap.
+    FeatureMap unique_features = feature_map;
+    unique_features.clear(false);
+
+    // A set of features disguised as a std::map.
+    // Uniqueness criterion/key is a pair <charge, sequence> for each feature.
+    typedef std::map<std::pair<Int, AASequence>, const Feature*> FeatureSet;
+    FeatureSet feature_set;
+
+    // 1. create a std::map `feature_set` mapping pairs <charge, sequence> to a pointer to 
+    // the feature with the highest intensity for this sequence.
+    for (FeatureMap::const_iterator fm_it = feature_map.begin(); fm_it != feature_map.end(); ++fm_it) 
+    {
+      const std::vector<PeptideIdentification> pep_ids = fm_it->getPeptideIdentifications();
+
+      if (!pep_ids.empty())
+      {
+        if (pep_ids.size() != 1) 
+        {
+          throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, __FUNCTION__, "Features may contain at most one identification. Run IDConflictResolver first to remove ambiguities!");
+        }
+
+        // Assumption: first hit returned by `getHits()` has always highest search engine score.
+        // TODO: Is this assumption reasonable or do we need to sort the hits first?
+        const PeptideHit highest_score_hit = pep_ids.begin()->getHits().front();
```

Comment:
first hit might not exist

Reply:
copy required?

Reply:
A `PeptideHit` without first hit? - Ok, we will test.

Reply:
sure - there can be PeptideIdentifications without hits